### PR TITLE
Add support for BBCode code style

### PR DESCRIPTION
**Pull request by blurymind**
This makes it possible to do [bb][/bb] code tag style.
To use bbcode style, add this line to the style:
```
	"default": {
		tagStyle: "bbcode"
	},
```

### CODE REVIEW

**Comment by blurymind** [comment]
Now it can also do most bbcode tags out of the box when using the `tagStyle: ["[", "]"]` without any regressions to the old functionality. They can also take parameters like `[color=red]red text[/color]`

**Comment by tleunen** [comment]
This is very interesting. I'm not against using a different syntax for this.
I wonder if in the future, we should focus on a single syntax instead of supporting more than one.

Any reason why you prefer the bracket syntax in comparison to `<>`?

**Comment by blurymind** [comment]
I actually forked this project in order to implement a bbcode parser for
pixijs. There is currently not a single one out there. The reason I needed
it for pixi is gdevelop. I am writing a bbcode object for it l and it uses
pixi for rendering

https://github.com/4ian/GDevelop/pull/1247

Why bbcode? It's more user friendly for non programmers and I am also using
if for yarn, which is now bundled with gdevelop

https://github.com/YarnSpinnerTool/YarnEditor

I am the main developer of yarn now :)

Anyways, I like the library for its flexibility. When forking it, I did
this in the spirit of retaining that spirit of flexibility

**Comment by blurymind** [comment]
Thank you for reviewing this. I will change it to enum and polish the or a bit more :)

**Comment by tleunen** [comment]
gdevelop is a good argument ;) Please assign me as reviewer when it's ready and I'll make sure to spend some more time reviewing this.

**Comment by blurymind** [comment]
@tleunen thank you :) 
I don't think I have the permission to assign reviewers to PRs on this tracker, but I can ping you with a comment when it's ready?

**Comment by blurymind** [comment]
@tleunen I replaced the tag style declarations with enums :)
Can you review this? I think I am getting closer to getting my bbcode object ready for gdevelop. Its feature complete now

**Comment by blurymind** [comment]
I don't mind :) I will add it at the end. Now when testing it extensively in gdevelop, I found a way to crash it - so I pushed a fix for that

Btw how did you make the html demo file? When I edit it in vscode,prettier moves the lines. I have to add it carefully with prettier turned off

**Comment by tleunen** [comment]
It was probably before prettier 😅 You could use "Save without formatting" in vscode (or similar editors) to prevent prettier to run too.

**Comment by blurymind** [comment]
updated demo now to show bbcode tags :) I added a couple of custom ones (spacing, outline and shadow).. I couldn't figure out a way to properly do the [u]underlined[/u] tag with pixi

**Comment by blurymind** [comment]
btw once approved, you have to do the merge, since I have no permission to do merges on this tracker :)

File: demo/index.html [hunk 1]
```diff
@@ -189,10 +199,11 @@ <h2>Have Fun</h2>
 			<div id="pixi-container">
 				<script>
 PIXI.settings.RESOLUTION = 2;
-let renderer = PIXI.autoDetectRenderer(600, 3100);
+let renderer = PIXI.autoDetectRenderer(600, 3425);
 renderer.backgroundColor = 0x333333;
 document.getElementById("pixi-container").appendChild(renderer.view);
 let stage = new PIXI.Container();
+const MultiStyleText = this.pixiMultistyleText;
```
Comment (tleunen) [inline demo/index.html]: Why is this needed? This is making the demo to fail on my machine. 

Not sure why the name would become `pixiMultistyleText`. The name used in microbundle is `MultiStyleText`.

Comment (blurymind) [reply demo/index.html]: ah sorry,I had to add it to get it to load on linux. I will remove it now

**Comment by blurymind** [comment]
> Everything looks good, but before merging and releasing a new version. I'd like to better understand the reason why we use `this.pixiMultistyleText`.

For some reason the demo import wasnt working on linux for me. When doing console.log(this), I found the library available under that namespace for some reason. Anyways, I removed that line from the pr now, as it wasn't really a part  of the feature - just a temporary way for me to play with the demo

**Comment by tleunen** [comment]
Thank you for this feature @blurymind!

**Comment by blurymind** [comment]
Thank you for taking the time to review it and also for developing this @tleunen 
Gdevelop users will really benefit from this and we will hopefully see it in many more pixijs games out there

**Comment by 4ian** [comment]
Thanks @blurymind for working on this and @tleunen for reviewing it (and making pixi-multistyle-text in the first place :)). GDevelop users will be happy to be able to use this!
